internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -867,203 +755,258 @@ func (b *BeatsMonitor) injectMetricsInput(
+			"type":       "system/metrics",
 			useOutputKey: monitoringOutput,
 			"data_stream": map[string]interface{}{
 				"namespace": monitoringNamespace,
 			},
-			"streams": beatsStreams,
+			"streams": []interface{}{
+				map[string]interface{}{
+					idKey: fmt.Sprintf("%s-%s", monitoringMetricsUnitID, name),
+					"data_stream": map[string]interface{}{
+						"type":      "metrics",
+						"dataset":   dataset,
+						"namespace": monitoringNamespace,
+					},
+					"metricsets":              []interface{}{"process"},
+					"period":                  metricsCollectionIntervalString,
+					"index":                   fmt.Sprintf("metrics-elastic_agent.%s-%s", name, monitoringNamespace),
+					"process.pid":             compPid,
+					"process.cgroups.enabled": false,
+					"processors":              processorsForProcessMetrics(name, comp.ID, monitoringNamespace, dataset, b.agentInfo),
+				},
+			},
+		}
+		inputs = append(inputs, input)
+	}
+	return inputs
+}
+
+// processorsForProcessMetrics returns processors used for process metrics.
+func processorsForProcessMetrics(binaryName, unitID, namespace, dataset string, agentInfo info.Agent) []any {
+	return []any{
+		addDataStreamFieldsProcessor(dataset, namespace),
+		addEventFieldsProcessor(dataset),
+		addElasticAgentFieldsProcessor(binaryName, agentInfo),
+		addAgentFieldsProcessor(agentInfo.AgentID()),
+		addComponentFieldsProcessor(binaryName, unitID),
+	}
+}
+
+// processorsForBeatsStream returns the processors used for metric streams in the beats input.
+func processorsForBeatsStream(binaryName, unitID, namespace, dataset string, agentInfo info.Agent) []any {
+	return []any{
+		addDataStreamFieldsProcessor(dataset, namespace),
+		addEventFieldsProcessor(dataset),
+		addElasticAgentFieldsProcessor(binaryName, agentInfo),
+		addAgentFieldsProcessor(agentInfo.AgentID()),
+		addComponentFieldsProcessor(binaryName, unitID),
+	}
+}
+
+// processorsForBeatsStream returns the processors used for metric streams in the beats input.
+func processorsForHttpStream(binaryName, unitID, dataset string, agentInfo info.Agent) []any {
+	sanitizedName := sanitizeName(binaryName)
+	return []interface{}{
+		addEventFieldsProcessor(dataset),
+		addElasticAgentFieldsProcessor(sanitizedName, agentInfo),
+		addAgentFieldsProcessor(agentInfo.AgentID()),
+		map[string]interface{}{
+			"copy_fields": map[string]interface{}{
+				"fields":         httpCopyRules(),
+				"ignore_missing": true,
+				"fail_on_error":  false,
+			},
 		},
 		map[string]interface{}{
-			idKey:        fmt.Sprintf("%s-agent", monitoringMetricsUnitID),
-			"name":       fmt.Sprintf("%s-agent", monitoringMetricsUnitID),
-			"type":       "http/metrics",
-			useOutputKey: monitoringOutput,
-			"data_stream": map[string]interface{}{
-				"namespace": monitoringNamespace,
+			"drop_fields": map[string]interface{}{
+				"fields": []interface{}{
+					"http",
+				},
+				"ignore_missing": true,
 			},
-			"streams": streams,
 		},
```

Comment:
Nit: I realize these two processors are not reused elsewhere, so that's why you probably didn't extract their creation into separate functions like you did for the other processors.  I do really like how readable the processor list has become with the function calls, since the function names now tell us the _purpose_ of the processor.  So I'd personally be +1 to extracting these two two processor definitions into their own functions as well.

Reply:
Sure, can do.

Reply:
Turns out they were reused, I just missed it. Good catch.